### PR TITLE
api: Add Sort::isInstantiated().

### PR DESCRIPTION
**Pull request by aniemetz**
This further removes Sort::isUinterpretedSortParameterized().

### CODE REVIEW

**Review by 4tXJ7f — COMMENTED** [review]
Should this method be marked experimental?

**Comment by aniemetz** [comment]
> Should this method be marked experimental?

Actually, no. And `Sort::instantiate` shouldn't either. I have included that fix in this PR.

File: src/api/cpp/cvc5.h [hunk 1]
```diff
@@ -593,6 +593,18 @@ class CVC5_EXPORT Sort
    */
   bool isUninterpretedSortConstructor() const;
 
+  /**
+   * Is this an instantiated (parametric datatype or unintpreted sort
```
Comment (ajreynol) [inline src/api/cpp/cvc5.h:597]:
```suggestion
   * Is this an instantiated (parametric datatype or uninterpreted sort
```

File: src/api/cpp/cvc5.h [hunk 2]
```diff
@@ -593,6 +593,18 @@ class CVC5_EXPORT Sort
...
+   * constructor) sort?
+   *
+   * An instantiated sort is a sort that has been constructed from
+   * instantiating sort parameters of a parametric sort with sort arguments
```
Comment (ajreynol) [inline src/api/cpp/cvc5.h:601]:
```suggestion
   * instantiating a sort with sort arguments
```

Comment (ajreynol) [reply src/api/cpp/cvc5.h:601]: Technically the uninterpreted sort constructor doesn't have parameters.

**Review by ajreynol — CHANGES_REQUESTED** [review]
A few minor comments else LGTM

**Review by ajreynol — APPROVED** [review]

**Pull request merged**